lib/chef/resource/yum_package.rb

```diff
@@ -62,6 +62,21 @@ class YumPackage < Chef::Resource::Package
       property :allow_downgrade, [ true, false ], default: false
 
       property :yum_binary, String
+
+      private
+
+      def get_package_name(x)
+        if is_valid_package?(x)
+          Chef::Log.error("Invalid package name #{x}")
+          exit 0
+        else
+          x.is_a?(Array) ? x.to_a : x
+        end
+      end
+
+      def is_valid_package?(x)
+        (x.is_a?(Array) && x.any? { |package_name| package_name.strip.index(/\s/) }) || (x.is_a?(String) && x.strip.index(/\b \b/))
```

Comment:
it is probably enough to fail on `.any? { |p| p =~ /\s/ }` or something like that, since package names should never contain whitespace (and the validation message should be something like that) spaces, tabs, newlines, etc are all not allowed.